src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,279 @@
+                    }
+
+                    var changedDocuments = projectChanges.SelectMany(pc => pc.GetChangedDocuments());
+                    var changedAnalyzerConfigDocuments = projectChanges.SelectMany(pc => pc.GetChangedAnalyzerConfigDocuments());
+                    var changedAdditionalDocuments = projectChanges.SelectMany(pc => pc.GetChangedAdditionalDocuments());
+
+                    // TO-DO: If the change involves modifying any document besides the document where the code action
+                    // was invoked, temporarily execute via command instead of WorkspaceEdit until LSP bug is fixed:
+                    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // After bug is fixed, remove the below if-statement and the existing code should work.
+                    if (changedDocuments.Any(d => d != document.Id) ||
+                        changedAnalyzerConfigDocuments.Any() ||
+                        changedAdditionalDocuments.Any())
+                    {
+                        return textDocumentEdits.ToArray();
+                    }
+
+                    // Changed documents
+                    foreach (var docId in changedDocuments)
+                    {
+                        var newDoc = applyChangesOperation.ChangedSolution.GetDocument(docId);
+                        var oldDoc = solution.GetDocument(docId);
+                        if (oldDoc == null || newDoc == null)
+                        {
+                            continue;
+                        }
+
+                        await GetTextDocumentEdits(textDocumentEdits, newDoc, oldDoc, cancellationToken).ConfigureAwait(false);
+                    }
+
+                    // Changed analyzer config documents
+                    // This for loop won't currently execute until https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // is fixed.
+                    foreach (var docId in changedAnalyzerConfigDocuments)
+                    {
+                        var newDoc = applyChangesOperation.ChangedSolution.GetAnalyzerConfigDocument(docId);
+                        var oldDoc = solution.GetAnalyzerConfigDocument(docId);
+                        if (oldDoc == null || newDoc == null)
+                        {
+                            continue;
+                        }
+
+                        await GetTextDocumentEdits(textDocumentEdits, newDoc, oldDoc, cancellationToken).ConfigureAwait(false);
+                    }
+
+                    // Changed additional documents
+                    // This for loop won't currently execute until https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // is fixed.
+                    foreach (var docId in changedAdditionalDocuments)
+                    {
+                        var newDoc = applyChangesOperation.ChangedSolution.GetAdditionalDocument(docId);
```

Comment:
these three loops are all doing basically the same thing, could we combine them?  Maybe a generic GetTextDocumentEdits(IEnumberable<T> documents, Func<DocumentId, T> GetChangedDocFunc)